app/models/relationship.rb

```diff
@@ -0,0 +1,7 @@
+class Relationship < ActiveRecord::Base
+  belongs_to :follower, class_name: 'User'
+  belongs_to :leader, class_name: 'User'
+  validates :follower_id, presence: true
+  validates :leader_id, presence: true
```

Comment:
I don't think these validations are OK. The main purpose of validations is to limit users inputs and ensure data integrity that way. In this case, its your app who is setting and creating new Relationship, so its like putting constraints on yourself, this is something you can do with tests instead.